src/main/java/edu/cornell/kfs/rass/batch/RassProjectDirectorConverter.java

```diff
@@ -1,74 +1,101 @@
 import org.kuali.kfs.krad.util.ObjectUtils;
 import org.kuali.kfs.module.cg.businessobject.CGProjectDirector;
+import org.kuali.kfs.sys.KFSConstants;
 import org.kuali.rice.kim.api.identity.Person;
 import org.kuali.rice.kim.api.identity.PersonService;
+import org.kuali.rice.kim.api.role.Role;
+import org.kuali.rice.kim.api.role.RoleService;
 
 import edu.cornell.kfs.rass.batch.xml.RassXMLAwardPiCoPiEntry;
 
 public class RassProjectDirectorConverter extends RassValueConverterBase {
+    private static final Logger LOG = LogManager.getLogger(RassProjectDirectorConverter.class);
 
-	protected PersonService personService;
-	protected Class<? extends CGProjectDirector> projectDirectorImplementationClass;
-	protected String projectDirectorPrimaryIndicatorPropertyName;
-
-	@SuppressWarnings("unchecked")
-	@Override
-	public Object convert(Class<? extends PersistableBusinessObject> businessObjectClass, RassPropertyDefinition propertyMapping, Object propertyValue) {
-		List<RassXMLAwardPiCoPiEntry> rassAwardPiCoPiEntries = (List<RassXMLAwardPiCoPiEntry>) propertyValue;
-		List<CGProjectDirector> projectDirectors = new ArrayList<>();
-		if (rassAwardPiCoPiEntries != null && rassAwardPiCoPiEntries.size() > 0) {
-			for (RassXMLAwardPiCoPiEntry rassAwardPiCoPi : rassAwardPiCoPiEntries) {
-				CGProjectDirector projectDirector = createNewProjectDirectorInstance();
-				Person projectDirectorPerson = personService
-						.getPersonByPrincipalName(rassAwardPiCoPi.getProjectDirectorPrincipalName());
-				if (ObjectUtils.isNull(projectDirectorPerson)) {
-				    throw new RuntimeException("Cannot find person with principal name \""
-				            + rassAwardPiCoPi.getProjectDirectorPrincipalName() + "\"");
-				}
-				projectDirector.setPrincipalId(projectDirectorPerson.getPrincipalId());
-				ObjectPropertyUtils.setPropertyValue(
-				        projectDirector, projectDirectorPrimaryIndicatorPropertyName, getNullSafePrimaryDirectorFlag(rassAwardPiCoPi));
-				projectDirectors.add(projectDirector);
-			}
-		}
-
-		return projectDirectors;
-	}
+    protected PersonService personService;
+    protected RoleService roleService;
+    protected Class<? extends CGProjectDirector> projectDirectorImplementationClass;
+    protected String projectDirectorPrimaryIndicatorPropertyName;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Object convert(Class<? extends PersistableBusinessObject> businessObjectClass, RassPropertyDefinition propertyMapping, Object propertyValue) {
+        List<RassXMLAwardPiCoPiEntry> rassAwardPiCoPiEntries = (List<RassXMLAwardPiCoPiEntry>) propertyValue;
+        List<CGProjectDirector> projectDirectors = new ArrayList<>();
+        if (rassAwardPiCoPiEntries != null && rassAwardPiCoPiEntries.size() > 0) {
+            for (RassXMLAwardPiCoPiEntry rassAwardPiCoPi : rassAwardPiCoPiEntries) {
+                CGProjectDirector projectDirector = createNewProjectDirectorInstance();
```

Comment:
may be cleaner to move the code inside the for loop to a separate method

Reply:
OK